cordova-lib/src/plugman/platforms/ubuntu.js

```diff
@@ -29,6 +29,54 @@ function toCamelCase(str) {
     }).join('');
 }
 
+function getPluginXml(plugin_dir) {
+    var et = require('elementtree'),
+    fs = require('fs'),
+    path = require('path');
+
+    var pluginxml;
+    var config_path = path.join(plugin_dir, 'plugin.xml');
+
+    if (fs.existsSync(config_path)) {
+        // Get the current plugin.xml file
+        pluginxml = et.parse(fs.readFileSync(config_path, 'utf-8'));
+    }
+ 
+    return pluginxml;
+}
+
+function findClassName(pluginxml, plugin_id) {
+    var class_name;
+
+    // first check if we have a class-name parameter in the plugin config
+    if (pluginxml) {
+	var platform = pluginxml.find("./platform/[@name='ubuntu']/");
+	if (platform) {
+	    var param = platform.find("./config-file/[@target='config.xml']/feature/param/[@name='ubuntu-package']");
+	    if (param && param.attrib) {
+		class_name = param.attrib.value;
+		return class_name;
+	    }
+	}
+    }
+
+    // fallback to guess work, based on the plugin package name
+
+    if (plugin_id.match(/\.[^.]+$/)) {
+        // old-style plugin name
+        class_name = plugin_id.match(/\.[^.]+$/)[0].substr(1);
+        class_name = toCamelCase(class_name);
+    } else {
+	match = plugin_id.match(/cordova\-plugin\-([\w\-]+)$/);
+        if (match && match.length > 0)
+	    class_name = match[0].substr(15);
+	else
+            class_name = toCamelCase(class_name);
+    }
+
+    return class_name;
+}
```

Comment:
Thanks @david-barth-canonical! Some comments:

* Generally the Cordova codebase uses curly braces even around single line blocks.
* There are some tabs in there that should be converted to spaces.
* The `match` variable needs a `var`.
* Should be calling `toCamelCase()` for the scenario where we match `cordova-plugin-...`
* Where we don't match `cordova-plugin-...`, `class_name` is not defined (should be using `plugin_id`).
* Since we will always end up calling `toCamelCase()`, the logic for the second half of this method can be simplified to:

```js
    // fallback to guess work, based on the plugin package name
    if (plugin_id.match(/\.[^.]+$/)) {
        // old-style plugin name
        class_name = plugin_id.match(/\.[^.]+$/)[0].substr(1);
    } else {
        var match = plugin_id.match(/cordova\-plugin\-([\w\-]+)$/);
        if (match && match.length > 0) {
            class_name = match[0].substr(15);
        } else {
            class_name = plugin_id;
        }
    }

    return toCamelCase(class_name);
```

Also, since the original PR has been merged, can you recreate this as a new PR which will just include the changes in your last commit (and changes in response to feedback, of course).